Delayed Account Creation: Remove experimental flag to enable feature

mikejolley opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR enables delayed account creation in the production build. The testing instructions enclosed cover new functionality added whilst behind the experimental flag. If you're testing this PR, test after using `npm run build`. If you're testing the release, just follow the instructions as-is.

Closes #50597

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Non-Block Themes**

1. Use a non-block theme such as StoreFront. 
2. Go to WooCommerce > Settings > Accounts and Privacy
3. The _Account Creation - After checkout (recommended)_ setting is not visible.

**Block Themes**

1. Use a block theme such as Twenty Twenty Five.
2. Go to WooCommerce > Settings > Accounts and Privacy
3. The _Account Creation - After checkout (recommended)_ setting is visible.
4. Toggle off "enable guest checkout"
5. Confirm that the "After checkout (recommended)" option is is greyed out and hovering over its checkbox shows a tooltip.
6. Toggle on "enable guest checkout"
8. Toggle off all "Account creation" settings
9. Confirm the "Account creation options" (Send password setup link) settings are greyed out

**Block Hooks**

1. WooCommerce > Settings > Accounts and Privacy - turn on account creation options and enable guest checkout.
2. Go to Appearance > Templates, find "order confirmation" in the list and click the menu (three dots) by its preview. Reset changes if the option is there. If its not, skip this step.
3. In the preview, ensure the account creation block is shown (you will see a "create account" button quite prominent in the preview).
4. In another browser, logged out as a guest, add something to cart and go through the checkout flow. Ensure you checkout with an email you have not used before.
5. On the order confirmation page, confirm you see the create account section. There will be some default text on the left listing benefits, and a button on the right. There may be a password field depending on settings too. Leave this tab open for testing. Do not create an account yet.
6. Back in the site editor, edit the order confirmation template. When inside, make some changes to the account creation block; edit text, headings, and make some styling changes e.g. background color. Save the template.
7. Back in the guest browser, refresh and confirm the changes are reflected on the frontend. There should also only be one create account block visible at all times.
8. Back as admin, Go to WooCommerce > Settings > Accounts and Privacy. Turn off  _Account Creation - After checkout (recommended)_ and save changes.
9. Back in the guest browser, refresh and confirm the account section is no longer shown.

**Customer facing functionality**

1. Ensure the feature is enabled in settings: registration and _Account Creation - After checkout (recommended)_  is enabled, and a block theme is in use.
2. Under WooCommerce > Settings > Accounts and Privacy, "Send password setup link (recommended)" is enabled.
3. Logged out as a guest, add something to cart and go through the checkout flow. 
4. On the order confirmation page, confirm you see the create account section. 
5. Click the "create account" button.
6. The page will load and show a confirmation that your account was created. Confirm the links in this section work
7. Check your email inbox for the registration email/new account email.
8. Check you can access the "my account" page and the order you placed is visible under the orders section.
10.  Under WooCommerce > Settings > Accounts and Privacy, "Send password setup link (recommended)" is disabled.
11. Repeat steps 3-8, but this time ensure the confirmation page prompts you to set a secure password before creating an account. Ensure you checkout with a different guest email or the create account section will be hidden.
12. Logout and login again using the password you set.
13. Logout and place another order as a guest using the _same_ email as before. This time the create account section will not be visible on the order confirmation page. 

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

mikejolley committed:
Remove experimental flags

mikejolley committed:
changelog

mikejolley committed:
Check theme is block theme

mikejolley committed:
adjust test since this setting is only visible with block themes

nielslange reviewed (approved):
Thanks for working on this, @mikejolley. I've successfully tested the changes and pre-approving this PR.

That said, I did spot the following things, which are not blocking this PR from getting merged:

- The testing instructions in `Block Hooks → 2.` say *"the preview, ensure the account creation block is shown"*. However, due to the previous instruction in `Block Themes → 7.`, which says, "Toggle off all "Account creation" settings"*, the account creation block is not visible by default. I suggest updating the testing instructions of the `Block Hooks` section, to ensure that testers activate *all "Account creation" settings.*
- The testing instructions in `Block Hooks → 3.` say *"On the order confirmation page, confirm you see the create account section."*. This will only work when using an email address that hasn't been registered already. I ran my first test using `admin@example.com`, but did not see the create account section. When testing again with an email address that hasn’t been registered yet, I could see the create account section. Of course, this behaviour makes sense. But I think it’s helpful to highlight this in the testing instructions.
- I noticed that when selecting the `Create account` button in the `Order Confirmation template`, a `Dark mode inputs` toggle appears, but it does not seem to affect the button. Toggling it, does not apply any changes.
- I noticed that both the text *“Create account”* and *“We'll email you a link ...”*, cannot be adjusted. Was there any discussion around that? At least for the button text *“Create account”*, I believe it’s useful to enable merchants to adjust this text directly. For the *“Proceed to Checkout”* and the *“Place Order”* buttons, we already enable merchants to edit the texts directly.

mikejolley committed:
Hide dark mode option if password input is disabled

mikejolley commented:
Thanks I've updated instructions.

> I noticed that when selecting the Create account button in the Order Confirmation template, a Dark mode inputs toggle appears, but it does not seem to affect the button. Toggling it, does not apply any changes.

This only affects inputs, like the toggle on checkout. 

I've changed it to only appear if the field is visible:
https://github.com/woocommerce/woocommerce/pull/52554/commits/8fa8c230dcf81a852c70722434592f316b5e4651

> I noticed that both the text “Create account” and “We'll email you a link ...”, cannot be adjusted. Was there any discussion around that? At least for the button text “Create account”, I believe it’s useful to enable merchants to adjust this text directly. For the “Proceed to Checkout” and the “Place Order” buttons, we already enable merchants to edit the texts directly.

I think this is a good suggestion but it wasn't in the designs. Only the left side content was editable. Logged a follow up which can be prioritised based on feedback.

https://github.com/woocommerce/woocommerce/issues/52669

Thanks
